eclipse-theia/theia
# Split off from git package the stuff provided by the git builtin plugin

fixes https://github.com/eclipse-theia/theia/issues/7152

Currently when using the VS Code git builtin, all the extra stuff available in Theia is no longer available.  For example and 'amend' section disappears, as does the history view.

#### What it does

This PR solves that.  The Git contribution to the Source Control view is moved to another package.  This allows the git package to be used alongside the builtin.

#### How to test

1. Include the `@theia/git` and `@theia/git-native` packages and check it works as before.  Check that you should see the 'amend' section and the history view.

2. Include `@theia/git` and the VS Code builtin plugin.  Check that the plugin is fully working and other plugins that depend on the git builtin also work.  Check that you get the 'amend' section and the history view.

One issue is that the Source Control view is slow to react to changes effected from the 'amend' section.  I don't know if there is some way of prodding the Git builtin to refresh.
  
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
